Core/DISET/private/BaseClient.py

```diff
@@ -227,35 +245,28 @@ def __discoverExtraCredentials(self):
           -> if KW_DELEGATED_GROUP in kwargs or delegatedGroup in threadConfig, put it in self.kwargs
           -> If we have a delegated DN but not group, we find the corresponding group in the CS
 
+        :return: S_OK()/S_ERROR()
     """
     # Wich extra credentials to use?
-    if self.__useCertificates:
-      self.__extraCredentials = self.VAL_EXTRA_CREDENTIALS_HOST
-    else:
-      self.__extraCredentials = ""
+    self.__extraCredentials = self.VAL_EXTRA_CREDENTIALS_HOST if self.__useCertificates else ""
     if self.KW_EXTRA_CREDENTIALS in self.kwargs:
       self.__extraCredentials = self.kwargs[self.KW_EXTRA_CREDENTIALS]
+
     # Are we delegating something?
-    delegatedDN, delegatedGroup = self.__threadConfig.getID()
-    if self.KW_DELEGATED_DN in self.kwargs and self.kwargs[self.KW_DELEGATED_DN]:
-      delegatedDN = self.kwargs[self.KW_DELEGATED_DN]
-    elif delegatedDN:
-      self.kwargs[self.KW_DELEGATED_DN] = delegatedDN
-    if self.KW_DELEGATED_GROUP in self.kwargs and self.kwargs[self.KW_DELEGATED_GROUP]:
-      delegatedGroup = self.kwargs[self.KW_DELEGATED_GROUP]
-    elif delegatedGroup:
-      self.kwargs[self.KW_DELEGATED_GROUP] = delegatedGroup
+    delegatedDN = self.kwargs.get(self.KW_DELEGATED_DN, self.__threadConfig.getDN())
```

Comment:
This does not behave the same if `self.kwargs[self.KW_DELEGATED_DN]` evaluates to `False`. Same bellow

Reply:
yes, it's true. fixed